src/app/qgisapp.cpp

```diff
@@ -8191,12 +8191,25 @@ void QgisApp::zoomOut()
 
 void QgisApp::zoomToSelected()
 {
-  mMapCanvas->zoomToSelected();
+  QList<QgsMapLayer *> layers = mLayerTreeView->selectedLayers();
+
+  if ( layers.size() > 1 && !layers.isEmpty() )
```

Comment:
```suggestion
  if ( layers.size() > 1 )
```

(can't be empty if it has > 1 items :wink: )